404.html

```diff
@@ -2,6 +2,15 @@
 layout: default
 ---
 
+<script>
+  let thePath = window.location.pathname;                   // Fetches the path (e.g. "/EIPS/eip-20-token-standard" or "/core")
+  let theRegex = /^(.*\/EIPS\/eip-\d+)[^\d\s\/][^\/]*$/gm;  // This regex matches the EIP path (e.g. "/EIPS/EIP-20") if the original path isn't valid
```

Comment:
While I don't think this is exploitable (because you use `pathname`), I am uncomfortable with a redirect to a location in the URL.